src/api/product-order/product-order.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { ProductOrder, ProductOrderSelect } from './model';
+import { ProductOrderArgs, ProductOrderCreateInput } from './dto';
+import { PrismaService } from 'src/shared/datasource/prisma/prisma.service';
+
+@Injectable()
+export class ProductOrderService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  public async findOne(
+    { where: productId_orderId }: ProductOrderArgs,
+    { select }: ProductOrderSelect,
```

Comment:
Validate that a product order can only be created as a regular user not a vendor, basically adding the Role decorator to the endpoint resolver.